### ISSUE TITLE
Importing types in JS files stopped working.

### ISSUE THREAD

**Issue by zavr-1** [Microsoft/TypeScript#27302]
Issue Type: <b>Bug</b>

There 

VS Code version: Code - Insiders 1.28.0-insider (2558a72d6902015b142ef45c57b664ce1f09e03f, 2018-09-21T05:12:46.853Z)
OS version: Darwin x64 15.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM)2 Duo CPU     P7350  @ 2.00GHz (2 x 2000)|
|GPU Status|2d_canvas: unavailable_software<br>checker_imaging: disabled_off<br>flash_3d: unavailable_software<br>flash_stage3d: unavailable_software<br>flash_stage3d_baseline: unavailable_software<br>gpu_compositing: unavailable_software<br>multiple_raster_threads: disabled_off<br>native_gpu_memory_buffers: unavailable_software<br>rasterization: unavailable_software<br>video_decode: unavailable_software<br>video_encode: unavailable_software<br>webgl: unavailable_off<br>webgl2: unavailable_off|
|Load (avg)|4, 11, 10|
|Memory (System)|8.00GB (0.12GB free)|
|Process Argv|/Applications/Visual Studio Code - Insiders.app/Contents/MacOS/Electron -psn_0_2023918|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (5)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|1.6.0
svg|joc|0.1.4
autoimport|ste|1.5.3
code-spell-checker|str|1.6.10
vscode-wakatime|Wak|1.2.3

</details>

Since last week, importing of external types has stopped working. 

![screen shot 2018-09-24 at 12 47 10](https://user-images.githubusercontent.com/21156791/45945983-06b32b00-bff8-11e8-85ec-0bf834d73f6b.png)

![screen shot 2018-09-24 at 12 52 03](https://user-images.githubusercontent.com/21156791/45946161-b9838900-bff8-11e8-8c16-e701c1c87b84.png)

![screen shot 2018-09-24 at 13 01 20](https://user-images.githubusercontent.com/21156791/45946651-32cfab80-bffa-11e8-9987-184df5aa7f3d.png)

It works in non-insiders. This has definitely worked in insiders fine last week. 

![screen shot 2018-09-24 at 12 48 03](https://user-images.githubusercontent.com/21156791/45946018-28acad80-bff8-11e8-92b3-a692e5f3a859.png)

![screen shot 2018-09-24 at 13 02 20](https://user-images.githubusercontent.com/21156791/45946657-382cf600-bffa-11e8-8349-1cbad8244a29.png)

And no I don't want to convert to TypeScript.

**Comment by sandersn** [comment]
My initial attempt at a repro failed -- everything succeeded. I'm going to try a clean `npm init` etc since I was testing in a directory with an explicit tsconfig.json and that already had `@types/koa` installed. This might be a failure in automatic type acquisition, since `@types/koa` has `app.middleware: compose.Middleware<koa.Context>[]` not `any[]`.

Do you have a tsconfig.json or jsconfig.json? Or have you explicitly done `npm install @types/koa`?

**Comment by sandersn** [comment]
From a fresh start, I can't repro your errors either; I think VS Code was either automatically installing `@types/koa` or perhaps it was in your package.json, because the types you have from your Working screenshot references types that are only in the d.ts, not in the actual package.

Specifically, `koa.Middleware` is a type defined only in the .d.ts.

```
✔ bug27302  [master|✚ 3…6]$ node
> const koa = require('koa')
undefined
> koa.Middleware
undefined
```

@andy-ms Does this look like a failure in Automatic Type Acquisition?

@zavr-1 Since this appears to be a problem with either the language service or VS Code, can you set the language service to dump its logs somewhere and send them to me or @andy-ms? You have to set this environment variable: `SET TSS_LOG=-level verbose -file C:\temp\logs\tsserver.log`. We'd like to see the log from setting up a new project that uses koa, then pasting in the simple repro you gave above.

My Microsoft mail is nathansa.

**Comment by andy-ms** [comment]
Adding `// @ts-check` to the file or `"checkJS": true` to a `tsconfig.json` or `jsconfig.json` will make ATA types inaccessible. Since you're getting red error underlines it seems like one of these must be set?

**Comment by zavr-1** [comment]
Hi no I don't have `tsconfig` or `jsconfig` I was just using running project as I always do and writing `type {import('koa').Middleware}` worked on Friday and then I came back after the weekend it stopped after a restart, so I reckoned it must be some commit that got merged...  also it's not just middleware it's everything.
Here's the log [tsserver.log](https://github.com/Microsoft/TypeScript/files/2412421/tsserver.log)

![test](https://user-images.githubusercontent.com/21156791/45972163-af3aac80-c043-11e8-9bb3-4cdc4a8a6337.gif)

**Comment by andy-ms** [comment]
I saw a crash when using vscode-insider's builtin TypeScript, but can't reproduce that in `typescript@next`. Could you try [setting up](https://code.visualstudio.com/Docs/languages/typescript#_using-newer-typescript-versions) vscode to use `typescript@next` and see if that fixes the problem?

**Comment by andy-ms** [comment]
Does it work if you `npm install @types/koa` explicitly?

**Comment by zavr-1** [comment]
And so it works today... Sorry don't know what's happened. The only thing different from last night was that there was a pending update (restart to update) to `2018-09-24T11:57:32.166Z` but it started working before I restarted. I guess it can be closed now...

**Comment by zavr-1** [comment]
Maybe it's another issue, but can you create a project, add `@idio/core` as dependency, write write

```js
import idioCore from '@idio/core'

(async () => {
  await idioCore()
})
```

Then jump into the `idioCore` -- will you have `koa-router` detected for types correctly? I don't. 

![screen shot 2018-09-25 at 16 08 46](https://user-images.githubusercontent.com/21156791/46016387-5586c080-c0dd-11e8-852a-aa42c9927894.png)

But I think this is because of https://github.com/Microsoft/TypeScript/issues/26921

**Comment by JacksonKearl** [comment]
Getting this as well, simple repro:

VSCode 1.27.2
TypeScript 3.0.3

package.json:
```
{
    "dependencies": {
        "express": "4.16.3"
    }
}
```

`npm install`

main.ts:
```
import * as express from "express" // Error: Could not find a declaration file for module 'express'. '/Users/jacksonkearl/Desktop/test/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

let a = express()

```

tsconfig.json:
```
{
    "compilerOptions": {
        "strict": true
    },
    // No change with this enabled:
    // "typeAcquisition": {
    //     "enable": true
    // }
}
```

**Comment by peteroid** [comment]
I was experiencing this issue while using the vscode with default Typescript version 3.1.0. Finally got the intellisense back after downgrading the ts to 3.0.3

**Comment by andy-ms** [comment]
@JacksonKearl typeAcquisition is a JS feature; in TS code you need to `npm install @types/express` like the message says.

**Comment by andy-ms** [comment]
@zavr-1 A related fix was just merged in #28236 -- I'm not 100% sure this will fix your issue, so please try out tomorrow's `typescript@next` release and tell us if this fixes it.
